src/mavlink_stream.c

```diff
@@ -96,16 +102,16 @@ void mavlink_stream(uint32_t time_us)
   {
     if (mavlink_streams[i].period_us && time_us - mavlink_streams[i].last_time_us >= mavlink_streams[i].period_us)
     {
-      mavlink_streams[i].last_time_us = time_us;
+      // if we took too long, set the last_time_us to be where it should have been
+      mavlink_streams[i].last_time_us = time_us - (time_us - mavlink_streams[i].last_time_us-mavlink_streams[i].period_us);
```

Comment:
So far, I haven't had that problem.  I had it streaming at 1000 Hz, and it wasn't any problem (any faster is stupid, because the IMU only reports at 1000 Hz).  The param reporting at the beginning slowed down, but I was still sending commands, receiving baro and mag, everything.  I think the mavlink-based system gives us _way_more efficient use of the bandwidth as compared with the old MSP system.